components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/service/CertificateValidationManagementServiceImpl.java

```diff
@@ -0,0 +1,219 @@
+            } else {
+                CACertificateInfo caCertificateInfo = new CACertificateInfo();
+                caCertificateInfo.setCertId(certificateId);
+                caCertificateInfo.setIssuerDN(getNormalizedName(caCertificate
+                        .getX509Certificate().getIssuerDN().getName()));
+                caCertificateInfo.setSerialNumber(getNormalizedName(caCertificate.getX509Certificate()
+                        .getSerialNumber().toString()));
+                caCertificateInfo.setCrlUrls(caCertificate.getCrlUrl());
+                caCertificateInfo.setOcspUrls(caCertificate.getOcspUrl());
+                return caCertificateInfo;
+            }
+        } catch (CertificateValidationException e) {
+            throw X509ConfigurationExceptionHandler
+                    .handleServerException(ErrorMessage.ERROR_WHILE_RETRIEVING_CA_CERTIFICATE_BY_ID, e);
+        }
+    }
+
+    @Override
+    public CACertificateInfo updateCACertificate(String certificateId, X509Certificate certificate, String tenantDomain)
+            throws X509ConfigurationException {
+
+        try {
+            CertObject certObject =
+                    updateCertificateInConfigurationStoreByCertificateId(certificateId, certificate, tenantDomain);
+            if (certObject == null) {
+                throw X509ConfigurationExceptionHandler
+                        .handleClientException(ErrorMessage.ERROR_CERTIFICATE_DOES_NOT_EXIST, certificateId,
+                                tenantDomain);
+            }
+            X509Certificate updatedCertificate = updateCACertificateInCertificateManager(certificateId,
+                    certificate, tenantDomain);
+            CACertificateInfo caCertificateInfo = new CACertificateInfo();
+            caCertificateInfo.setCertId(certificateId);
+            caCertificateInfo.setIssuerDN(getNormalizedName(updatedCertificate.getIssuerDN().getName()));
+            caCertificateInfo.setSerialNumber(getNormalizedName(updatedCertificate.getSerialNumber().toString()));
+            caCertificateInfo.setCrlUrls(certObject.getCrlUrls());
+            caCertificateInfo.setOcspUrls(certObject.getOcspUrls());
+            return caCertificateInfo;
+        } catch (CertificateValidationException | CertificateException | CertificateMgtException e) {
+            throw X509ConfigurationExceptionHandler
+                    .handleServerException(ErrorMessage.ERROR_WHILE_UPDATING_CA_CERTIFICATE, e);
+        }
+    }
+
+    @Override
+    public void deleteCACertificate(String certificateId, String tenantDomain) throws X509ConfigurationException {
+
+        try {
+            CertObject certObject = deleteCertificateInConfigurationStoreByCertificateId(certificateId, tenantDomain);
+            if (certObject != null) {
+                deleteCACertificateFromCertificateManager(certificateId, tenantDomain);
+            } else {
+                throw X509ConfigurationExceptionHandler
```

Comment:
Do we need to inform the client regarding the resource inexistence

Reply:
this is handled in api layer